core/src/test/java/io/questdb/cutlass/text/ParallelCsvFileImporterTest.java

```diff
@@ -0,0 +1,1991 @@
+        });
+    }
+
+    @Test
+    public void testImportOnlyOneActiveImport() throws Exception {
+        executeWithPool(8, 2, (CairoEngine engine, SqlCompiler compiler, SqlExecutionContext sqlExecutionContext) -> {
+            final String tableName = "tableName";
+            compiler.compile("create table " + tableName + " ( ts timestamp, line symbol, d double, description symbol) timestamp(ts) partition by MONTH;", sqlExecutionContext);
+            CountDownLatch latch = new CountDownLatch(1);
+            Thread importer = new Thread(() -> {
+                try (ParallelCsvFileImporter indexer = new ParallelCsvFileImporter(sqlExecutionContext)) {
+                    indexer.setMinChunkSize(10);
+                    indexer.of(tableName, "test-quotes-big.csv", PartitionBy.MONTH, (byte) ',', "ts", "yyyy-MM-ddTHH:mm:ss.SSSSSSZ", true);
+                    indexer.processTest(latch::countDown);
+                } catch (SqlException e) {
+                    e.printStackTrace();
+                }
+                Path.clearThreadLocals();
+            });
+
+            importer.start();
+            latch.await();
+
+            try (ParallelCsvFileImporter indexer = new ParallelCsvFileImporter(sqlExecutionContext)) {
+                indexer.setMinChunkSize(10);
+                indexer.of(tableName, "test-quotes-big.csv", PartitionBy.MONTH, (byte) ',', "ts", "yyyy-MM-ddTHH:mm:ss.SSSSSSZ", true);
+                try {
+                    indexer.process();
+                    Assert.fail();
+                } catch (SqlException e) {
+                    MatcherAssert.assertThat(e.getMessage(), containsString("Another parallel copy command in progress"));
+                }
+            }
+
+            importer.join();
+        });
+    }
+
+    @Test
+    public void testAssignPartitionsToWorkers() {
+        ObjList<PartitionInfo> partitions = new ObjList<>();
+
+        partitions.add(new PartitionInfo(1, "A", 10));
+        partitions.add(new PartitionInfo(2, "B", 70));
+        partitions.add(new PartitionInfo(3, "C", 50));
+        partitions.add(new PartitionInfo(4, "D", 100));
+        partitions.add(new PartitionInfo(5, "E", 5));
+
+        int tasks = ParallelCsvFileImporter.assignPartitions(partitions, 2);
+
+        MatcherAssert.assertThat(partitions, equalTo(new ObjList<PartitionInfo>(new PartitionInfo(1, "A", 10, 0),
+                new PartitionInfo(4, "D", 100, 0),
+                new PartitionInfo(5, "E", 5, 0),
+                new PartitionInfo(2, "B", 70, 1),
+                new PartitionInfo(3, "C", 50, 1))));
+        MatcherAssert.assertThat(tasks, equalTo(2));
+    }
+
+    static ObjList<IndexChunk> list(IndexChunk... chunks) {
+        ObjList<IndexChunk> result = new ObjList<>(chunks.length);
+        for (IndexChunk chunk : chunks) {
+            result.add(chunk);
+        }
+        return result;
+    }
+
+    static void executeVanilla(TestUtils.LeakProneCode code) throws Exception {
```

Comment:
nit: is this needed? it's called just from `executeWithPool()` anyway.

Reply:
fixed